### DISCUSSION TITLE
Solving phase-field fracture for chemo-mechanical problem with only thermal loading

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20119] [Q&A Modules: General] [answered]
Hello MOOSE experts

I am trying to solve the phase-field fracture for a chemo-mechanical environment with only flux loading.

I found that the following module doesn't include the Eigen strain due to which the stresses are not developing and fracture is not evolving due to the lack of sufficient driving force.

Can you please suggest any alternative to calculate all the following parameters while including the effect of Eigen strain? I shall be very thankful.

```
   type = ComputeLinearElasticPFFractureStress
    c = damage
    E_name = 'elastic_energy'
    D_name = 'degradation'
    F_name = 'local_fracture_energy'
    decomposition_type = stress_spectral  
#    decomposition_type = strain_vol_dev
    use_snes_vi_solver = true
  [../]
```

Best
Avtar

**Comment by hugary1995** [comment]
I doubt if stress-free configurations lead to fracture in the first place.

Beside that, you can use strain based decompositions in this case, so that the fracture driving force isn't zero. Since stress-free thermal eigenstrains should lead to nonzero mechanical strains.

**Reply by avtarsinghh1991** [reply]
Thank you very much.

Actually, stress itself is coming out to be zero.

I tried strain decomposition as well. It's not working.

Best
Avtar

**Reply by hugary1995** [reply]
Could you elaborate on what isn't working? For example, does it fail to converge? Does it result in no fracture? If it's the latter case, you may want to examine the state of the mechanical strain.

**Reply by avtarsinghh1991** [reply]
Hello @hugary1995 

I am trying to solve phase-field fracture in a chemo-mechanical environment. The diffusion is working absolutely fine. However, upon using the following module, the attendant stresses are found to be zero means stresses are not evolving even after taking into account the Eigen strain. As the stresses are zero, there is no driving force that evolves the fracture. Hence, the damage is also not evolving.


My concern is how can I take care of Eigen strain using the "ComputeLinearElasticPFFractureStress"?

Lets say in case of small strain

elastic_strain = total_strain - eigen_strain

It seems this part is somehow missing. How can I update the elastic_strain?



```
  [./elasticity_tensor]
    type = ComputeElasticityTensor
    C_ijkl = '121.15e9 80.769e9'
    fill_method = symmetric_isotropic
  [../]
    [./eigen_strain_prefactor]
    type = DerivativeParsedMaterial
    args = diffused
    f_name = eigen_strain_prefactor
    constant_names = 'c_ref'
    constant_expressions = '30000.0'
    function = (diffused-c_ref)/3.0
  [../]
  [./eigenstrain]
    type = ComputeVariableEigenstrain
    args = diffused
    eigen_base = '3.496e-6 0 0 0 3.496e-6 0 0 0 3.496e-6'
    prefactor = eigen_strain_prefactor
    eigenstrain_name = eigenstrain
  [../]

  [./define_mobility]
    type = ParsedMaterial
    material_property_names = 'gc_prop visco'
    f_name = L
    function = '1.0/(gc_prop * visco)'
  [../]
  [./define_kappa]
    type = ParsedMaterial
    material_property_names = 'gc_prop l'
    f_name = kappa_op
    function = 'gc_prop * l'
  [../]
  [./damage_stress]
    type = ComputeLinearElasticPFFractureStress
    c = damage
    E_name = 'elastic_energy'
    D_name = 'degradation'
    F_name = 'local_fracture_energy'
    decomposition_type = strain_spectral  
#    decomposition_type = strain_vol_dev
    use_snes_vi_solver = true
  [../]
  [./degradation]
    type = DerivativeParsedMaterial
    f_name = degradation
    args = 'damage'
    function = '(1.0-damage)^2*(1.0 - eta) + eta'
    constant_names       = 'eta'
    constant_expressions = '1.0e-5'
    derivative_order = 2
  [../]
  [./local_fracture_energy]
    type = DerivativeParsedMaterial
    f_name = local_fracture_energy
    args = 'damage'
    material_property_names = 'gc_prop l'
    function = 'damage^2 * gc_prop / 2 / l'
    derivative_order = 2
  [../]
  [./fracture_driving_energy]
    type = DerivativeSumMaterial
    args = damage
    sum_materials = 'elastic_energy local_fracture_energy'
    derivative_order = 2
    f_name = F
  [../] 
```


Best
Avtar

**Reply by hugary1995** [reply]
> My concern is how can I take care of Eigen strain using the "ComputeLinearElasticPFFractureStress"?

`ComputeLinearElasticPFFractureStress` does take into account eigenstrain if you use a strain based decomposition. If you do have non-zero elastic strain, and the elastic strain isn't in contraction state, there _will be_ a fracture driving force. So again, could you examine the state of the elastic strain output from your simulation?

**Comment by avtarsinghh1991** [comment, marked as answer]
Thank you very much. It has been resolved.
